src/power_grid_model_io/mappings/unit_mapping.py

```diff
@@ -58,7 +59,7 @@ def set_mapping(self, mapping: Units):
                     )
                 self._mapping[unit] = (multiplier, si_unit)
                 if unit == si_unit:
-                    if multiplier != 1.0:
+                    if not np.isclose(multiplier, 1.0, rtol=1.0e-9, atol=0.0):
```

Comment:
Why is it needed to do it this way?

Reply:
sonar cloud tells us that for instance parsing strings `int("1.0")` may still give rounding errors, so comparing floats is dangerous